src/ui-kit/experimental/hierarchical/hierarchical-tree-grid/hierarchical-tree-grid.component.ts

```diff
@@ -36,116 +35,149 @@ export interface GridItem {
   styleUrls: ['./hierarchical-tree-grid.component.scss']
 })
 export class SamHierarchicalTreeGridComponent implements OnInit {
+
+  /**
+  * Table configurations 
+  */
   @Input() public templateConfigurations: GridTemplateConfiguration;
-  @Input() public template: GridTemplate;
-  @Input() public dataSource: any;
 
-  @Output() public itemSelected = new EventEmitter<GridItem>();
+  /**
+  * Allow to insert a customized template for suggestions to use
+  */
+  @Input() public gridTemplate: GridTemplate;
+
+  /**
+  * Allow to search the data on the table.
+  */
+  @Input() public filterText: string;
+
+    /**
+  * Data for the Table.
+  *  Simple data array
+  * Stream that emit a array each time when the item is selected.
+  * Stream that changes each time when click action trigger on row.
+  */
+  @Input() public gridData: any[] = [];
+
+  /**
+ * Event emitted when level change is clicked
+ */
   @Output() public levelChanged = new EventEmitter<GridItem>();
-  public selectedItem: GridItem;
-  public chkSelected: boolean = false;
-  selectedItemIndex = 0;
 
-  public displayedColumns = ['select'];
-  public samTableDataSource: any | null;
-  dataChange: BehaviorSubject<any[]> = new BehaviorSubject<any[]>([]);
-  public focusedCell: any;
-  public focusedtemIndex = 0;
+  /**
+ * Event emitted when row is clicked
+ */
+  @Output() public rowChanged = new EventEmitter<GridItem>();
 
+  /**
+  * Event emitted when row set is selected.
+  */
+  @Output() selectResults = new EventEmitter<any[]>();
```

Comment:
@yerramshilpa Can you take a look at all your properties and make them more specific?